src/Microsoft.Azure.WebJobs.Extensions.Storage/DeadlockHandler.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.ExceptionServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Timers;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Storage
+{
+    internal static class DeadlockHandler
+    {
+        private static readonly TimeSpan DefaultDeadlockTimeout = TimeSpan.FromMinutes(2);
+
+        public static async Task<T> RestartIfDeadlocked<T>(string operationName, IWebJobsExceptionHandler exceptionHandler, Func<Task<T>> operation)
```

Comment:
This technically just implements a timeout pattern, not deadlock detection, right? You're doing the same thing here that you do for TimeoutAttribute in FunctionExecutor. So perhaps this helper should be renamed to ExecuteWithTimeout

Reply:
Yeah, I started renaming it with my other solution -- which was actually tracking and killing the deadlocked Response. I'll update this one to be more generically-named.